### DISCUSSION TITLE
Event to Command Behavior Not Working?

### DISCUSSION THREAD

**Discussion by EmberLynn** [CommunityToolkit/Maui#533] [Q&A]
I'm trying to use the EventToCommandBehavior on an Entry field. My code is as follows:

![eventtocommand](https://user-images.githubusercontent.com/46328589/182719863-3a0128b6-22aa-46ce-9f83-23484b123c2c.png)

I know that TextChanged is the event that fires whenever something is typed into the Entry. However, I cannot get the ToggleIsSubmitButtonEnabledCommand to fire (It does work, I can fire it through a Command Property on a button). Does anyone have any advice as to why the Community Toolkit behavior won't work on this? 

P.S. I removed the TextChanged="OnTextFieldChanged" from the Entry, but that did not solve the problem.

**Comment by VladislavAntonyuk** [comment]
You need to pass context to the command. Sample: https://github.com/xamarin/XamarinCommunityToolkit/blob/main/samples/XCT.Sample/Pages/Behaviors/CharactersValidationBehaviorPage.xaml#L57

**Reply by EmberLynn** [reply]
Thanks for your response, but could you please elaborate? I am new to MAUI and app development, in general. What context am I passing to the command and how do I pass that context? 

Here is the documentation for this specific command: 
https://docs.microsoft.com/en-us/dotnet/communitytoolkit/maui/behaviors/event-to-command-behavior

**Reply by pictos** [reply]
@EmberLynn, is this project on GitHub? That way we can take a look at it.

**Reply by EmberLynn** [reply]
I apologize, this isn't available on github. This is a company project I'm working on.

**Reply by pictos** [reply]
So, from your screenshot, I believe the `ContentPage.BindingContext` is `AddEditRequestMockupViewModel`. So, `Behaviors` don't inherit the `BindingContext` from its parent, you need to do it.

```xml
<ContentPage  x:Name= "this">

<Entry>
   <Entry.Behaviors>
          <t: EventToCommandBehavior Command="{Binding YourCommand, Source={x:reference this.BindingContext}}" />
   <Entry.Behaviors>
</Entry>
</ContentPage>
```
So, here's a snippet of the code, I think this will work for you

**Reply by EmberLynn** [reply]
Thank you for your recommendation. Unfortunately, I was not able to get it to work. Here is what my code looks like after making the suggested change:
```XAML
<DataTemplate x:Key="EntryFieldTemplate">
    <VerticalStackLayout Padding="5"
                         x:DataType="model:CustomFormField">
        <Label Text="{Binding Name}"/>
        <Entry Placeholder="Enter your sentence here"
               Text="{Binding TextValue}"
               BindingContextChanged="OnTextFieldChanged">
            <Entry.Behaviors>
                <toolkit:EventToCommandBehavior
                    EventName="TextChanged"
                    Command="{Binding ToggleIsSubmitButtonEnabledCommand, Source={x:Reference this.BindingContext}}"/>
            </Entry.Behaviors>
        </Entry>
    </VerticalStackLayout>
</DataTemplate>
```
Also, I included x:Name="this" to reference the ContentPage as you suggested.

**Reply by VladislavAntonyuk** [reply]
try something like this: https://github.com/VladislavAntonyuk/MauiSamples/blob/main/MauiBank/Views/CardPage.xaml#L102

if it doesn't work please include a small repro

**Comment by CliffAgius** [comment]
Closed as answered
